src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs

```diff
@@ -46,6 +46,8 @@ internal sealed override CommandLineArguments CommonParse(IEnumerable<string> ar
         /// <returns>a commandlinearguments object representing the parsed command line.</returns>
         public new CSharpCommandLineArguments Parse(IEnumerable<string> args, string baseDirectory, string sdkDirectory, string additionalReferenceDirectories = null)
         {
+            Debug.Assert(baseDirectory == null || PathUtilities.IsAbsolute(baseDirectory));
```

Comment:
>IsAbsolute [](start = 64, length = 10)

Regarding the assertion of absolute path, where does this guarantee come from?
Since this API is public, the user could pass any path they want. Please add a test for that.

Does the rest of our code have a strong assumption of an absolute path? If so, we should be reporting an error to users. Otherwise, we should remove the assertion.

Reply:
Parse() calls into FlattenArgs(), which calls into ResolveRelativePath(path, baseDirectory). ResolveRelativePath currently contains an [identical assertion](http://source.roslyn.io/#Microsoft.CodeAnalysis/FileSystem/FileUtilities.cs,99) using PathUtilities.IsAbsolute.

Basically, the CommandLineParser has always assumed the baseDirectory would be an absolute path. This new assert is supposed to make the assumption clear to the reader.

Since it's a public API, maybe just a Debug.Assert is not quite right? Would it be better to have the method always throw on a non-absolute baseDirectory and to have a test that ensures it throws?

Reply:
Since assertions are skipped in release mode, there is also the question of how the underlying code behaves with a relative path. Maybe it works and some users took a dependency on it.

I'm fine with leaving the assertion as-is. 
If the underlying code blows up on relative paths, then changing what exception we produce doesn't seem a great improvement.